haystack_experimental/components/extractors/llm_metadata_extractor.py

```
@@ -0,0 +1,172 @@
+    #############################
+    -Real Data-
+    ######################
+    entity_types: [company, organization, person, country, product, service]
+    text: {{input_text}}
+    ######################
+    output:
+    '''
+
+    docs = [
+        Document(content="deepset was founded in 2018 in Berlin, and is known for its Haystack framework"),
+        Document(content="Hugging Face is a company founded in Paris, France and is known for its Transformers library")
+    ]
+
+    extractor = LLMMetadataExtractor(prompt=NER_PROMPT, expected_keys=["entities"])
+    extractor.run(documents=docs)
+    >> {'documents': [
+        Document(id=.., content: 'deepset was founded in 2018 in Berlin, and is known for its Haystack framework',
+        meta: {'entities': [{'entity': 'deepset', 'entity_type': 'company'}, {'entity': 'Berlin', 'entity_type': 'city'},
+              {'entity': 'Haystack', 'entity_type': 'product'}]}),
+        Document(id=.., content: 'Hugging Face is a company founded in Paris, France and is known for its Transformers library',
+        meta: {'entities': [
+                {'entity': 'Hugging Face', 'entity_type': 'company'}, {'entity': 'Paris', 'entity_type': 'city'},
+                {'entity': 'France', 'entity_type': 'country'}, {'entity': 'Transformers', 'entity_type': 'product'}
+                ]})
+           ]
+       }
+    >>
+    ```
+    """ # noqa: E501
+    def __init__(
+            self,
+            prompt: str,
+            expected_keys: List[str],
+            model: Optional[str] = None,
+            raise_on_failure: bool = True
+    ):
+        self.prompt = prompt
+        self.builder = PromptBuilder(prompt)
```

Could we also set the `required_variables` param here to avoid silent failures in case for some reason the `input_text` isn't passed at run time?
```suggestion
        self.builder = PromptBuilder(prompt, required_variables=["input_text"])
```